Python/Product/PythonTools/PythonTools/Project/PytestConfigureInfoBar.cs

```diff
@@ -0,0 +1,222 @@
+                getPackagesTask.Add(packageManager.GetInstalledPackagesAsync(CancellationToken.None));
+            }
+
+            Boolean pytestPackageFound = false;
+            var packages = await Task.WhenAll(getPackagesTask);
+            foreach (var packageSpecs in packages) {
+                var result = packageSpecs.FirstOrDefault(
+                    x => String.Equals(x.Name, nameof(TestFrameworkType.Pytest), StringComparison.OrdinalIgnoreCase
+                ));
+
+                if (result != null) {
+                    pytestPackageFound = true;
+                    break;
+                }
+            }
+
+            if (pytestPackageFound) {
+                ShowInfoBar(
+                    new InfoBarHyperlink(Strings.ConfigurePytestInfoBarEnableAction, (Action)EnablePytest),
+                    pyTestConfigFilePath,
+                    caption
+                );
+            } else {
+                ShowInfoBar(
+                    new InfoBarHyperlink(Strings.ConfigurePytestInfoBarInstallAndEnableAction, (Action)InstallAndEnablePytest),
+                    pyTestConfigFilePath,
+                    caption
+                );
+            }
+        }
+
+        private void ShowInfoBar(InfoBarActionItem configureActionItem, string pytestConfigFilePath, string caption) {
+            var messages = new List<IVsInfoBarTextSpan>();
+            var actions = new List<InfoBarActionItem>();
+
+            messages.Add(new InfoBarTextSpan(
+                Strings.ConfigurePytestTxtInfoBarMessage.FormatUI(
+                    PathUtils.GetFileOrDirectoryName(pytestConfigFilePath),
+                    caption
+            )));
+
+            actions.Add(configureActionItem);
+            actions.Add(new InfoBarHyperlink(Strings.ConfigurePytestTextInfoBarIgnoreAction, (Action)Ignore));
+
+            LogEvent(ConfigurePytestInfoBarActions.Prompt);
+            Create(new InfoBarModel(messages, actions, KnownMonikers.StatusInformation));
+        }
+
+        private async void InstallAndEnablePytest() {
+            LogEvent(ConfigurePytestInfoBarActions.EnableAndInstall);
+
+            //IS there a better way to do this? Combining two link queries? TODO
+            Boolean pytestInstalled = false;
+            var packageManagers = _interpreterOptionsService.GetPackageManagers(_interpreterFactory);
+            foreach (var packageManager in packageManagers) {
+                pytestInstalled = await packageManager.InstallAsync(
+                    new PackageSpec("pytest"),
+                    new VsPackageManagerUI(Site),
+                    CancellationToken.None
+                    );
+
+                if (pytestInstalled) {
+                    SetProperty(PythonConstants.TestFrameworkSetting, "Pytest");
+                    break;
+                }
+            }
+
+            if (!pytestInstalled) {
+                //Pytest installation failed.Show error message? TODO
```

Comment:
installing will print to output window, so the error message will already be there, so you could force show the output window.